zip-0315.rst

```diff
@@ -1,9 +1,629 @@
+desired transactions to occur, and exposing any privacy implications to the user so
+that they have enough information to assess the consequences of their economic actions.
+
+This ZIP covers best practices for:
+
+* what information to display to the user about transactions and balances;
+* how to handle interactions between the ZIP 32 [#zip-0032]_ key tree and Unified Addresses [#zip-0316]_;
+* when to use external or internal keys/addresses;
+* sharing addresses and viewing keys;
+* sending and receiving funds (including fee selection);
+* migrating funds between pools.
+
+
+Requirements
+============
+
+User consent
+------------
+
+A wallet typically reveals some information in the process of creating a transaction.
+Which information is revealed depends on the configured wallet privacy policy.
+The guiding principle of these requirements is that users must explicitly consent
+to each instance of information being revealed by the wallet in a transaction.
+
+A user may give blanket consent to reveal a particular kind of information, and
+must also be able to change the configured wallet privacy policy to avoid the
+wallet creating new information leaks of a given type.
+
+The specifications below describe some situations in which blanket consent may be
+inappropriate.
+
+Some varieties of consent may not be revocable, for example if a user chooses to
+share some of their keys.
+
+
+Prompt accessibility of funds
+-----------------------------
+
+Wallets need to take account of two concerns:
+
+* enabling funds to be spent as quickly as possible to reduce latency;
+* waiting long enough before spending TXOs to ensure that the
+  confirmed-spendable balance is not overestimated, and so can be trusted
+  by the user.
+
+To enable this we define two kinds of TXOs:
+
+* A trusted TXO is a TXO received from a party where the wallet trusts a
+  double-spend not to occur, e.g. TXOs created by the wallet's internal TXO
+  handling.
+* An untrusted TXO is a TXO received by the wallet that is not trusted.
+
+
+Wallets can then require that untrusted TXOs need more confirmations before
+they become spendable than trusted TXOs. This provides an improved trade-off
+between latency on the one hand, and reliability and safety on the other.
+
+
+Specification
+=============
+
+Long-term storage of funds
+--------------------------
+
+It is RECOMMENDED that wallets only hold funds as shielded in the long term;
+that is, if a wallet supports receiving transparent funds (or supports
+importing a seed from another wallet that might have done so), then it SHOULD
+either auto-shield or opportunistically shield such funds by default.
+
+Following this recommendation improves both collective privacy and the user's
+individual privacy, by maximizing the size of the note anonymity set over time.
+
+The remainder of this specification assumes a wallet that follows the above
+recommendation, except where explicitly noted.
```

Comment:
```suggestion

If a shielding transaction spends from multiple transparent addresses, those
addresses will be linked. Despite the fact that it is not possible to achieve strong
privacy guarantees from any use of transparent addresses, it is undesirable to
reveal this additional information. Both auto-shielding and opportunistic shielding
SHOULD therefore only shield from funds held in a single transparent address,
unless the user has given explicit consent to do otherwise.

Following these recommendations improves both collective privacy and the user's
individual privacy, by maximizing the size of the note anonymity set over time and
minimizing the potential linkage of shielding transactions to other transactions.

The remainder of this specification assumes a wallet that follows the above
recommendations, except where explicitly noted.
```